Completion/Unix/Command/_stow

```diff
@@ -0,0 +1,89 @@
+#compdef stow chkstow
+
+#
+# A zsh completion script for GNU stow (https://www.gnu.org/software/stow/)
+#
+
+__stow_complete_packages() {
+  local stow_dir=${(Q)1}
+  local -a stow_pkg_list=( $stow_dir/*(-/N:t) )
+
+  if [[ ${#stow_pkg_list} -gt 0 ]]; then
+    _values -C "packages from $stow_dir" ${stow_pkg_list[@]}
+  else
+    _message "no packages found in $stow_dir"
+  fi
+}
+
+case $service in
+  stow)
+    local context state line curcontext="$curcontext" ret=1
+    typeset -A opt_args
+    # Others local variables
+    local stow_dir arguments
+
+    arguments=(
+      '(- *)'{--help,-h}'[show help]'
+      '(- *)'{--version,-V}'[show version number]'
+      '(-d --dir)'{-d+,--dir=}'[set the stow dir (default is current dir)]:stow dir:_files -/'
+      '(-t --target)'{-t+,--target=}'[set the target dir (default is parent of stow dir)]:target dir:_files -/'
+      '*'{-S,--stow}'[stow the package names that follow]:stow package:->stow_package'
```

Comment:
Looking at the `stow` man page, i don't think these specs are right:

https://www.gnu.org/software/stow/manual/stow.html#Mixing-Operations

Off the top of my head i'm not sure what would be the ideal way to do this. You could use the `:*:...` mechanism, but there's no way to return from it. idk. At the least, you might consider leaving a note here that it's not accurate

Also, the `:stow package:` in all of these is superfluous (and misleading)

Reply:
Indeed, I hadn't thought of this case. But after a few tests, this doesn't seem to be a problem for stow, even if it can lead to surprising results... I have added a note to explain that point.

However, I do not understand the following comment:

> Also, the :stow package: in all of these is superfluous (and misleading)

if I remove `:stow package:` it doesn't work anymore...

Reply:
Try just `: :`.

The description is ignored because it goes to the stow_package state.

Reply:
done in 0ebc9639e